src/extension/src/utils/dependencyChecker.ts

```diff
@@ -0,0 +1,75 @@
+import { WizardServant, IPayloadResponse } from "../wizardServant";
+import { ExtensionCommand } from "../constants";
+const os = require('os');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const PYTHON27 = 'Python 2.7';
+
+export class DependencyChecker extends WizardServant {
+  clientCommandMap: Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  >;
+
+  constructor() {
+    super();
+    this.clientCommandMap = this.defineCommandMap();
+  }
+
+  private defineCommandMap(): Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  > {
+    return new Map([[ExtensionCommand.CheckDependency, this.checkDependency]]);
+  }
+
+  private async checkForPython3() {
+    var state;
+    try {
+      const { stdout } = await exec('py -3 --version'); // using Python Launcher command
+      if (stdout.length > 0) {
+        state = true;
+      }
+    } catch (err) {
+      state = false;
+    }
+    return state;
+  }
+
+  private outputIsPython27(stdout: any, stderr: any) {
+    return stdout.indexOf(PYTHON27) === 0 || stderr.indexOf(PYTHON27) === 0;
+  }
+
+  async checkDependency(message: any): Promise<IPayloadResponse> {
+    var userOS = os.platform();
+    var userOnWin = userOS.indexOf('win') === 0;
+    var name = message.payload.dependency === 'python' && !userOnWin ? 'python3'
+                                                                     : message.payload.dependency; 
+    var state;
+    try {
+      const { stdout, stderr } = await exec(name + ' --version');
```

Comment:
Security concern: should we sanitize the name?
What if someone posts malicious command?
At the other hand, if the message cannot come from outside of extension client, maybe it's not an issue.

@jcoc611-microsoft , thoughts?

Reply:
How do we sanitize the name? 

This part of the code can only be triggered through the client's posted messages, so no messages can come from outside of the client.

Reply:
Seems alright for now, as it can only be one of the PYTHON constants